### ISSUE TITLE
IncompatibleClassChangeError Exception v1.4.0 in 2022.2 EAP 3

### ISSUE THREAD

**Issue by sabihoshi** [seclerp/rider-efcore#81]
#### Describe the bug

**To Reproduce**

1. Create a new console project
2. Right click on the project
3. See an error generated by the plugin

### Environment (please complete the following information):

- OS: Windows 11 Pro Insider Preview 22H2 25131.1000
- .NET SDK info (installed ones and version used within Rider): .NET 6.0.3
- Plugin version: v1.4.0
- Rider version: 2022.2 EAP 3 Build #RD-222.2889.22, built on June 4, 2022

#### Additional context
```java
update failed for AnAction(me.seclerp.rider.plugins.efcore.features.migrations.add.AddMigrationAction, id=EfCore.Features.Migrations.AddMigrationAction) with text=Add Migration

java.lang.IncompatibleClassChangeError: Found interface com.jetbrains.rider.projectView.workspace.ProjectModelEntity, but class was expected
	at me.seclerp.rider.plugins.efcore.features.shared.AnActionEventExKt.getDotnetProject(AnActionEventEx.kt:30)
	at me.seclerp.rider.plugins.efcore.features.shared.AnActionEventExKt.isEfCoreActionContext(AnActionEventEx.kt:16)
	at me.seclerp.rider.plugins.efcore.features.shared.EfCoreAction.update(EfCoreAction.kt:24)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performDumbAwareUpdate$0(ActionUtil.java:139)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:155)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:655)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$updateActionReal$4(ActionUpdater.java:128)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$6(ActionUpdater.java:214)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:66)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$7(ActionUpdater.java:212)
	at com.intellij.openapi.actionSystem.impl.ActionUpdateEdtExecutor.lambda$computeOnEdt$0(ActionUpdateEdtExecutor.java:45)
	at com.intellij.openapi.actionSystem.impl.Utils.lambda$expandActionGroupFastTrack$5(Utils.java:289)
	at com.intellij.openapi.actionSystem.impl.Utils.runLoopAndWaitForFuture(Utils.java:687)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroupFastTrack(Utils.java:287)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroupImpl(Utils.java:200)
	at com.intellij.openapi.actionSystem.impl.Utils.lambda$fillMenu$6(Utils.java:313)
	at com.intellij.openapi.actionSystem.impl.Utils.computeWithRetries(Utils.java:725)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:312)
	at com.intellij.openapi.actionSystem.impl.ActionMenu.fillMenu(ActionMenu.java:434)
	at com.intellij.openapi.actionSystem.impl.ActionMenu.setPopupMenuVisible(ActionMenu.java:379)
	at java.desktop/javax.swing.JPopupMenu.menuSelectionChanged(JPopupMenu.java:1509)
	at java.desktop/javax.swing.MenuSelectionManager.setSelectedPath(MenuSelectionManager.java:134)
	at java.desktop/javax.swing.plaf.basic.BasicMenuUI.appendPath(BasicMenuUI.java:264)
	at java.desktop/javax.swing.plaf.basic.BasicMenuUI$Actions.actionPerformed(BasicMenuUI.java:322)
	at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:311)
	at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:243)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

**Comment by seclerp** [comment]
Thanks for that, seems like something changed between EAP releases that caused runtime incompatibility. I will look into this.

**Comment by sabihoshi** [comment]
Thank you, I just tested it and it works on 2022.2 EAP 3, it generated the zip file and I was able to import it and now I am able to use it again.

Kind of weird there's so many errors and some severe warnings in the frontend.log but still proceeded to do fine, similar to the workflow run https://github.com/seclerp/rider-efcore/runs/6804296677?check_suite_focus=true#step:4:3594 here. Otherwise, this issue is now solved.
